packages/openapi-fetch/test/index.test.ts

```diff
@@ -1352,6 +1352,33 @@ describe("client", () => {
         customProperty: "value",
       });
     });
+
+    it("multipart/form-data with a file", async () => {
+      const TEST_STRING = "Hello this is text file string";
+
+      const file = new Blob([TEST_STRING], { type: "text/plain" });
+      const formData = new FormData();
+      formData.append("file", file);
+
+      const client = createClient<paths>({ baseUrl });
+      useMockRequestHandler({
+        baseUrl,
+        method: "post",
+        path: "/multipart-form-data-file-upload",
+        handler: async (data) => {
+          // Get text from file and send it back
+          const formData = await data.request.formData();
+          const text = await (formData.get("file") as File).text();
+          return new HttpResponse(JSON.stringify({ text }));
+        },
+      });
+      const { data } = await client.POST("/multipart-form-data-file-upload", {
+        // TODO: how to get this to accept FormData?
+        body: formData as unknown as string,
```

Comment:
How to get this to accept FormData?

Reply:
Oh hm do we need to adjust the types here? Is there an improvement we could make in `packages/openapi-fetch/index.d.ts` for `body`?

Reply:
Oh this will be tricky. This library’s whole purpose is to assert types for `body` based on your OpenAPI schema. https://github.com/openapi-ts/openapi-typescript/blob/main/packages/openapi-fetch/src/index.d.ts#L90

We don’t want to _always_ allow FormData as that would mess up some people’s code. So we want to conditionally allow FormData in some instances where it’s possible (e.g. not for GET). So we’ll need to surgically add FormData as a union into request bodies, but only when it may be possible/desired.

Given the complexity, I’m OK with merging this PR without that just to unblock the runtime, and the types can be tackled separately (sidenote: that’s why the runtime and types are separated—the type complexity isn’t possible to express with normal runtime code)

Reply:
What are your thoughts? Do you want to try and solve the type errors? Or merge by requiring manual overrides?

Reply:
I'm ok with merging now. I think letting FormData through the body serializer is a nice little addition in itself. Oh, I was wondering why the types are separated, that's very cool to know 👍